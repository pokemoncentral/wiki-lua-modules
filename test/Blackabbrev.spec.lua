-- "Test cases" for Blackabbrev

local ba = require("Blackabbrev")

-- A very long one, with both black and white text
local solid = ba.RuZaSmRFVF({ args = {} })
print(solid)

-- The very long one, but split
local split = ba.RuZa({ args = { "Sm", "RFVF" } })
print(split)

-- Syntax equivalence

assert(solid == split, "syntax mismatch")

-- No color specified

local noColor = ba.LGPE({ args = {} })
print(noColor)

-- _abbr function

local abbr = ba._abbr({ args = { "RBG RFVF LGPE" } })
print(abbr)

-- Test Ranger

local ranger1 = ba.Ra({ args = {} })
print(ranger1)

local ranger2 = ba.OSA({ args = {} })
print(ranger2)

local ranger3 = ba.TDL({ args = {} })
print(ranger3)
