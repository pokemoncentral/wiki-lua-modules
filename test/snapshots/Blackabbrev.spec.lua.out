[[Pokémon Rubino e Zaffiro|<span style="padding: 0 0.3em; background: #A00000; color: #EAEAEA;">'''R'''</span><span style="padding: 0 0.3em; background: #0000A0; color: #EAEAEA;">'''Z'''</span>]][[Pokémon Smeraldo|<span style="padding: 0 0.3em; background: #00A000; color: #EAEAEA;">'''S'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="padding: 0 0.3em; background: #FF7327; color: #000;">'''RF'''</span><span style="padding: 0 0.3em; background: #00DD00; color: #000;">'''VF'''</span>]]
[[Pokémon Rubino e Zaffiro|<span style="padding: 0 0.3em; background: #A00000; color: #EAEAEA;">'''R'''</span><span style="padding: 0 0.3em; background: #0000A0; color: #EAEAEA;">'''Z'''</span>]][[Pokémon Smeraldo|<span style="padding: 0 0.3em; background: #00A000; color: #EAEAEA;">'''S'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="padding: 0 0.3em; background: #FF7327; color: #000;">'''RF'''</span><span style="padding: 0 0.3em; background: #00DD00; color: #000;">'''VF'''</span>]]
[[Pokémon: Let's Go, Pikachu! e Let's Go, Eevee!|<span style="padding: 0 0.3em; background: #EAEAEA; color: #000;">'''LG'''</span><span style="padding: 0 0.3em; background: #ffcf11; color: #000;">'''P'''</span><span style="padding: 0 0.3em; background: #c47e39; color: #EAEAEA;">'''E'''</span>]]
[[Pokémon Rosso e Blu|<span style="padding: 0 0.3em; background: #FF1111; color: #EAEAEA;">'''R'''</span><span style="padding: 0 0.3em; background: #1111FF; color: #EAEAEA;">'''B'''</span>]][[Pokémon Giallo|<span style="padding: 0 0.3em; background: #FFD733; color: #000;">'''G'''</span>]][[Pokémon Rosso Fuoco e Verde Foglia|<span style="padding: 0 0.3em; background: #FF7327; color: #000;">'''RF'''</span><span style="padding: 0 0.3em; background: #00DD00; color: #000;">'''VF'''</span>]][[Pokémon: Let's Go, Pikachu! e Let's Go, Eevee!|<span style="padding: 0 0.3em; background: #EAEAEA; color: #000;">'''LG'''</span><span style="padding: 0 0.3em; background: #ffcf11; color: #000;">'''P'''</span><span style="padding: 0 0.3em; background: #c47e39; color: #EAEAEA;">'''E'''</span>]]
[[Pokémon Ranger (gioco)|<span style="padding: 0 0.3em; background: #F7681A; color: #EAEAEA;">'''Ra'''</span>]]
[[Pokémon Ranger: Ombre su Almia|<span style="padding: 0 0.3em; background: #2D4B98; color: #EAEAEA;">'''OsA'''</span>]]
[[Pokémon Ranger: Tracce di luce|<span style="padding: 0 0.3em; background: #2CB8E9; color: #000;">'''Tdl'''</span>]]
